eva/udfs/asl_action_recognition.py

```diff
@@ -0,0 +1,103 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle as pkl
+
+import numpy as np
+import pandas as pd
+import torch
+import torchvision
+
+try:
+    from torchvision.models.video import R3D_18_Weights, r3d_18
+
+except ImportError:
+    raise ImportError(
+        f"torchvision>=0.14.0 is required to use MVITActionRecognition, found {torchvision.__version__}"
+    )
+import torch.nn as nn
+import torchvision
+
+from eva.models.catalog.frame_info import FrameInfo
+from eva.models.catalog.properties import ColorSpace
+from eva.udfs.abstract.pytorch_abstract_udf import PytorchAbstractClassifierUDF
+
+
+class ASLActionRecognition(PytorchAbstractClassifierUDF):
+    @property
+    def name(self) -> str:
+        return "ASLActionRecognition"
+
+    def download_weights(self):
+        if not os.path.exists(self.asl_weights_path):
+            torch.hub.set_dir(os.getcwd())
+            torch.hub.download_url_to_file(
+                self.asl_weights_url,
+                self.asl_weights_path,
+                hash_prefix=None,
+                progress=True,
+            )
+
+    def setup(self):
+        self.asl_weights_url = (
+            "https://gatech.box.com/shared/static/crjhyy4nc2i5nayesfljutwc1y3bpw2q.pth"
+        )
+        self.asl_weights_path = os.getcwd() + "/asl_weights.pth"
```

Comment:
Consider using this https://github.com/georgia-tech-db/eva/blob/04040765a43bc43e699feacb461d3b063bcbfb08/eva/configuration/constants.py#L30

Reply:
updated.